### ISSUE TITLE
Ability to specify version in the directory path of the regex

### ISSUE THREAD

**Issue by shijinabraham** [concourse/s3-resource#51]
I am trying to use s3 resource to store build artifacts in directory which are named according to the versions. I am trying the following 

s3 resource regexp is "regexp:  server_build/release-(.*)/output.txt"
s3 put is  " file: output/release-*/output.txt"

Running this task gives this error.

panic: version number was not valid: Invalid character(s) found in major number "("

**Comment by vito** [comment]
I don't really know what to do with this. From what's there I don't see where the `(` would come from. Did you have a weird filename or something?

**Comment by shijinabraham** [comment]
No. The filename and path is what is given (output.txt). The version is just semver.  When I change the version to be append to the file, it works. But when I try to append version to directory, this error is returned.

**Comment by vito** [comment]
Ah, yeah that's broken at the moment; the resource expects the final path segment (the filename) to contain the version.

**Comment by loganmzz** [comment]
18 months later, what's status on this issue ? We also want to group up all transient artifacts of same version under a unique folder.

**Comment by kalfa** [comment]
same here, a hierarchical/versioned directory structure is quite useful.

dir-VER/
  - artefact1
  ...
  - artefactN

is preferible in many cases we have than either
dir/
 - artefact1-VER
 - artefactN-VER

or 

dir/
 - artefact1/data-VER
 - artefact2/data-VER

**Comment by eedwards-sk** [comment]
I'm also screwed by this right now, because I have a CI process that requires the filename be static (homebrew bottles), but I could work around that to create multiple 'versions' by generating them in different root urls.

e.g. instead of just `s3://my-bucket/my_static_filename_pattern.tar.gz` and thus concourse caching `my_static_filename_pattern.tar.gz` even though it has changed...

I could put them in `s3://my-bucket/my-dynamic-prefix/my_static_filename_pattern.tar.gz` and each 'path' would then result in a new version.

Honestly, considering s3 'folders' are actually part of the file name, I think we should be including the prefix in the version anyway.
